dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/ChangeLogController.java

```diff
@@ -538,6 +535,7 @@ public RootNode getTrackedEnityInstanceChangeLog(
   }
 
   @GetMapping("trackedEntity")
+  @Deprecated(since = "2.42", forRemoval = true)
```

Comment:
Why is this one deprecated and which endpoint should be used instead?

Reply:
This endpoint returns audit entries for the given TE. These entries can either be READ or SEARCH, depending on the type of request the user made to access a particular TE.

There are a few issues related to this. The first is that some of the READ entries are not created by a user accessing a TE but are instead generated by system processes or validations. These entries provide no value to the end user.
The second issue is related to security. Currently, there is no ACL validation on this endpoint, allowing a user to access the audit of a TE that is not within their search scope.
Moreover, neither Android nor Frontend are using this endpoint.

For all these reasons, we are deprecating this endpoint with no replacement in sight until we come up with a better design. There are ongoing discussions regarding the security issue, depending on the outcome, we might remove this endpoint completely.